App/LoginValidate.js

```diff
@@ -0,0 +1,113 @@
+var LOGIN_SUCCESFULLY = 0;
+var INVALID_PASSWORD =1;
+var INVALID_USERNAME =2;
+
+var usernameJson= JSON.parse(usersJSON);
+console.log(usernameJson);
+// Users class
+function User(data) {
+  this.name=data.userName;
+  this.email=data.email;
+  this.password=data.password;
+}
+//end User class
+
+
+// AccountService class
+function AccountService() {
+  this.username="";
+  this.password="";
+  this.usersArray=[];
+}
+AccountService.prototype.setUsername= function(username){
+  this.username=username;
+}
+AccountService.prototype.setPassword= function(password){
+  this.password=password;
+}
+AccountService.prototype.addUser= function(user){
+  this.usersArray.push(user);
+}
+AccountService.prototype.checkLogin = function(){
+  var toReturn = false;
+  for (var i = 0; i < this.usersArray.length; i++) {
+    var tempUser= this.usersArray[i];
+    console.log(tempUser.userName," ? ", this.username);
+    if(tempUser.userName==this.username)
+    {
+      if(tempUser.password==this.password){
```

Comment:
The code is too complicate here. Use a return statement instead of break and eliminate else branches